services/app-api/libs/authorization.js

```diff
@@ -7,29 +7,41 @@ import { SimpleJsonFetcher } from "aws-jwt-verify/https";
 
 export async function getUserDetailsFromEvent(event) {
```

Comment:
How would you feel about this change to getUserDetailsFromEvent?
```js
export async function getUserDetailsFromEvent(event) {
  await verifyEventSignature(event);

  if (event.requestContext.accountId === "000000000000") { 
    return {}
  }

  // TODO, it seems that jwtDecode and verifier.verify may return the same object?
  // Maybe we can remove the jwtDecode dependency.

  const token = jwtDecode(event?.headers?.["x-api-key"]);
  const role = mapMembershipToRole(token["custom:ismemberof"]);

  return {
    email: token.email,
    firstName: token.given_name,
    lastName: token.family_name,
    role,
    username: token.identities?.[0]?.userId || token.email,
    usernameSub: token.sub, // whatever that means.
  };
}
```

Reply:
everything I did so far to x-api-key was just awful, I need to work on it more and figure out why I had to do that to get login to work, what I did was wrong, very wrong.